### ISSUE TITLE
Supabase OAuth Callback setup not working

### ISSUE THREAD

**Issue by ndom91** [malerba118/supabase-comments-extension#2]
Hey, this might be a dumb question, but this is the first time I've used supabase and I can't get the oauth callback setup to work.

I've setup an oauth clientId / clientSecret in Github, and added `https://[projectId].supabase.co/auth/v1/callback` as the callback URL. I've also added the prop `redirectTo={'https://myblogpage.com'}` to the `AuthModal` component.

However, after clicking "Signin with github" in the AuthModal, and asked to allow my custom Github App, I'm redirected to that supabase URL with an accessToken and verification code appended, so far so good. Decoding the jwt accessToken also reveals my correct settings from supabase, but then the browser just says `Invalid Redirect`.

![image](https://user-images.githubusercontent.com/7415984/152659396-f07242e7-ade4-4e09-ba2d-2087c658d874.png)

Am I missing something obvious here? I'm assuming supabase is supposed to direct me back to the URL specified in the `redirectTo` prop. Right?

**Comment by ndom91** [comment]
Got it workined, issue was I had multiple URL's listed in the supabase dashboard's `Site URL` field.

![image](https://user-images.githubusercontent.com/7415984/152659606-167843d7-5b77-4302-abba-6a56a32e48f8.png)

If you can only have one there, I guess you have to manually change it back n forth between prod / dev while you're developing. Is that how you go about it?

**Comment by malerba118** [comment]
Hey man, not dumb at all, def a bit confusing. 

It sounds like what you've done is correct, i think you're just missing a whitelist of auth redirect urls on the supabase dashboard.

<img width="1421" alt="Screen Shot 2022-02-05 at 2 26 19 PM" src="https://user-images.githubusercontent.com/5760059/152659603-c3d2a24e-ab12-48e5-b13e-7aa37172f8bb.png">

**Comment by malerba118** [comment]
Ahh yeah, looks like you're on the right track. Basically what i've done (and i'm not sure if this is correct but it seems to work for me) is put my main prod site as the Site URL and then in additional redirect urls put localhost, codesandbox, any lower environment for example. I think if there's just one callback url then supabase knows unambiguously where to redirect the user, but if you have many then i think you have to use the redirectTo prop on the Auth component to disambiguate and as long as supabase finds the redirectTo url in your whitelist then it'll pass the user through successfully.

**Comment by ndom91** [comment]
Okay, yeah makes sense. Thanks for the quick response!

P.S. One other small suggestion - I see in your example you're manually adding the "Signout" link under the CommentProvider. But in the actual library, the signin link just gets replaced with "Submit" for the comment.

Might be useful to build in this part 

```js
!auth.user && (
  <button onClick={() => supabase.auth.signout()} />
)
```

into the component you're actually shipping :+1: 

Maybe just next to the submit btn? 

![image](https://user-images.githubusercontent.com/7415984/152659881-df0b1c53-a004-4c94-8c97-a8104df70495.png)

**Comment by malerba118** [comment]
Good idea, i think i'm into it. Definitely want to do as much as possible to make it work just out of the box. And for those who want more customization they can always add their own sign out button and call supabase.auth.signOut() themself.

Thinking something like this, like a button group with a dropdown menu and in that menu there'd be a sign out option.

<img width="181" alt="Screen Shot 2022-02-05 at 2 45 32 PM" src="https://user-images.githubusercontent.com/5760059/152660128-3c40a79d-8efb-4d6e-aa0d-39299164a278.png">

**Comment by ndom91** [comment]
Yeah I was thinking about just adding the signout button myself too, but it'd be hard to integrate it nicely into the preexisting comment box ui, right?

I could float it below, etc. but thats not really nice.

The combobox button isn't a bad choice either!

**Comment by malerba118** [comment]
Yeah i don't want the sign-out button to be too in the user's face is all, because when faced with two buttons "Submit" and "Sign Out", i think 95% of the time the user will be clicking submit
